src/ArrayInterface.jl

```diff
@@ -36,8 +36,26 @@ const MatAdjTrans{T,M<:AbstractMatrix{T}} = Union{Transpose{T,M},Adjoint{T,M}}
 const UpTri{T,M} = Union{UpperTriangular{T,M},UnitUpperTriangular{T,M}}
 const LoTri{T,M} = Union{LowerTriangular{T,M},UnitLowerTriangular{T,M}}
 
-@inline static_length(a::UnitRange{T}) where {T} = last(a) - first(a) + oneunit(T)
-@inline static_length(x) = Static.maybe_static(known_length, length, x)
```

Comment:
Technically this isn't part of the public API, but just in case it might be good to do `@deprecate static_length(x) = ArrayInterface.length(x)`.

Reply:
The JuliaSIMD ecosystem has been using `ArrayInterface.static_length`, e.g.
https://github.com/JuliaSIMD/LoopVectorization.jl/search?q=static_length
so I would prefer a major version bump and PRs to these packages to update them.

Reply:
Yeah, that's what I was afraid of